### PR TITLE
Added Java Solution to Egg Dropping Puzzle Problem assigned.

### LINKED ISSUE #107: Egg Dropping Puzzle
**This issue is up for Python implementation.**

Egg dropping refers to a class of problems in which it is important to find the correct response without exceeding a (low) number of certain failure states. 

**Is your feature request related to a problem? Please describe.**
In a toy example, there is a tower of n floors and an egg dropper with m ideal eggs. The physical properties of the ideal egg are such that it will shatter if it is dropped from floor n^*n or above, and will have no damage whatsoever if it is dropped from floor n^*-1n or below. The problem is to find a strategy such that the egg dropper can determine the floor n^*n in as few eggs drop as possible. 

**Describe the solution you'd like**
This problem has many applications in the real world such as avoiding a call out to the slow HDD or attempting to minimize cache misses or running a large number of expensive queries on a database.

**Additional context**
Content is taken from https://brilliant.org/wiki/egg-dropping/

**Languages**
- [x] Python
- [x] Java
- [ ] C
- [x] C++

### PR DESCRIPTION
**Pull request by theDeepanshuMourya**
# Added DP Solution to the problem assigned for Java Language.

Fixes #107 

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

# Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generates no new warnings

### CODE REVIEW

File: Java/EggDroppingPuzzle.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (kaustubhgupta) [inline Java/EggDroppingPuzzle.java:1]: You are deleting one file from the Java Directory, "CoinChange" Kindly resolve that

File: Java/MinimumPathSum.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
```
Comment (kaustubhgupta) [inline Java/MinimumPathSum.java]: Remove this unnecessary file

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by theDeepanshuMourya** [comment]
I've deleted the unnecessary file as suggested.

**Comment by kaustubhgupta** [comment]
> I've deleted the unnecessary file as suggested.

This PR will delete another contribution file. I would suggest making another PR, also update the Readme of JAVA folder

**Comment by YashSaxena9** [comment]
Please create a new PR for the issue, current PR is being closed because of merge conflicts.

**Comment by theDeepanshuMourya** [comment]
Please guide me on how to do that because I am new to all this. I am not able to make a pull request without the deleted file.

**Comment by kaustubhgupta** [comment]
@theDeepanshuMourya delete your forked repo, fork it again and then add the file which solves the issue. Plus update the readme of the Java folder with your entry. Don't worry it happens for the first time and it's good to know you are actively participating in it!

**Comment by theDeepanshuMourya** [comment]
Okay. Thanks. I'll do it as suggested.

**Pull request closed without merging**